### PR TITLE
Adds MergeSort Algorithm in Java

### PR DESCRIPTION
**Pull request by fedesassone**
Merge Sort is a sorting algorithm, which is commonly used in computer science. Merge Sort is a divide and conquer algorithm. It works by recursively breaking down a problem into two or more sub-problems of the same or related type, until these become simple enough to be solved directly. The solutions to the sub-problems are then combined to give a solution to the original problem. So Merge Sort first divides the array into equal halves and then combines them in a sorted manner.

Closes https://github.com/NITSkmOS/Algorithms/issues/149

**By submitting this pull request I confirm I've read and complied with the
below declarations.**

- [X] I have read the [Contribution guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [commit guidelines](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#write-good-commit-messages) for this project.
- [X] My code follows the [standard code structure](https://github.com/NITSkmOS/Algorithms/blob/master/CONTRIBUTING.md#samples-code).
- [X] This pull request has a descriptive title. For example, `{Tag}: Add {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [X] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.
- [X] This pull request shall only be reviewed and merged once all the checks passes. No maintainer or supporter shall be obliged to review it before this condition is met.
- [X] I have mentioned the issue number correctly (with hyperlink) in this pull request description.

After you submit your pull request, **DO NOT** click the 'Update Branch' button.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,7 @@ This repository contains examples of various algorithms written on different pro
...
 |:----------------------------------------------------------------------------------------------- |:-------------------------------------:|:-------------------------------------:|:-------------------------------------:|:-------------------------------------:|
 | [Euclidean GCD](https://en.wikipedia.org/wiki/Euclidean_algorithm)                              | [:octocat:](euclidean_gcd/C)          |                                      | [:octocat:](euclidean_gcd/Java)        | [:octocat:](euclidean_gcd/Python)     |
-| [Merge Sort](https://en.wikipedia.org/wiki/Merge_sort)                                          | [:octocat:](merge_sort/C)             |                                       |                                       | [:octocat:](merge_sort/Python)        |
+| [Merge Sort](https://en.wikipedia.org/wiki/Merge_sort)                                          | [:octocat:](merge_sort/C)             |                                       |[:octocat:](euclidean_gcd/Java)                                       | [:octocat:](merge_sort/Python)        |
```
Comment (sangamcse) [inline README.md]: euclidean_gcd ??

File: merge_sort/Java/merge_sort.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+import java.util.Scanner;
```
Comment (sangamcse) [inline merge_sort/Java/merge_sort.java]: Remove this imprt.

File: merge_sort/Java/merge_sort.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        System.out.println("The unsorted array is:");
+        System.out.println(Arrays.toString(intArray));
+
+        mergesort(intArray);
+
+        System.out.println("The sorted array is:");
+        System.out.println(Arrays.toString(intArray));
+    }
+
+    static void mergesort(int intArray[]) {
```
Comment (sangamcse) [inline merge_sort/Java/merge_sort.java]: Please add comments for this method. You can understand how much code comments are necessary for the maintenance.

File: merge_sort/Java/merge_sort.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        int left[] = new int[mid];
+        int right[] = new int[size - mid];
+        for (int i = 0; i < mid; i++) left[i] = intArray[i];
+        for (int i = mid; i < size; i++) right[i - mid] = intArray[i];
+        mergesort(left);
+        mergesort(right);
+        merge(intArray, left, right);
+    }
+
+    public static void merge(int array[], int left[], int right[]) {
```
Comment (sangamcse) [inline merge_sort/Java/merge_sort.java]: Comments please

Comment (fedesassone) [reply merge_sort/Java/merge_sort.java]: Done.

File: merge_sort/Java/merge_sort.java (hunk 3 above)
Comment (fedesassone) [reply merge_sort/Java/merge_sort.java]: Done.

**Comment by sangamcse** [comment]
ack 1f0414dd4a5fa6d8264abab08685882d8fc925e2

**Comment by sangamcse** [comment]
@sangamcse rebase

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated rebase with [GitMate.io](https://gitmate.io) was successful! :tada:

**Comment by sangamcse** [comment]
ack 48971af8813a42091495ece7f4bec4ec45963e2a

**Comment by sangamcse** [comment]
ack 2aca28e3ae4392267986a0ca9d38ecda9aefab0e

**Comment by sangamcse** [comment]
@sangamcse ff

**Comment by sangamcse** [comment]
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being fastforwarded automatically. Please **DO NOT** push while fastforward is in progress or your changes would be lost permanently :warning:

**Comment by sangamcse** [comment]
Automated fastforward with [GitMate.io](https://gitmate.io) was successful! :tada:

**Pull request merged**